Sources/Build/misc.swift

```diff
@@ -61,6 +61,16 @@ extension ClangModule {
 }
 
 extension Product {
+    /// Creates output path of a product by appending working directory and product outname.
+    ///
+    /// - Parameters:
+    ///     - prefix: Working directory in which product will be created.
+    ///
+    /// - Returns: Output path of the product.
+    func outpath(_ prefix: String) -> String {
```

Comment:
I'm ambivalent on the value of this extra method... I feel like in the current code base it doesn't add clarity.

If we wanted to make sure this was more nicely factored, I think we would do so by creating a set of "bound" products which had their paths assigned _once_ and then make all the code go through that. This is fundamentally different from having a method which takes the prefix, though, because the exact value would only be assigned once.